marlowe-playground-client/src/MainFrame/State.purs

```diff
@@ -274,8 +278,33 @@ handleAction s (HaskellAction action) = do
       selectView BlocklyEditor
     _ -> pure unit
 
+handleAction s (JavascriptAction action) = do
+  toJavascriptEditor (JavascriptEditor.handleAction s action)
+  case action of
+    JS.SendResultToSimulator -> do
+      mContract <-
+        peruse
+          ( _javascriptState
+              <<< JS._compilationResult
+              <<< JS._CompiledSuccessfully
+              <<< JSI._result
+          )
+      let
+        contract = case mContract of
+          Just con -> show $ pretty con
+          _ -> ""
```

Comment:
One idea for refactoring this would be to move the way you get the variable `contract `to the specific module (Javascript/Haskell), in this case everything between 285 to 295 extracted as a function in Javascript/State.purs, same for Haskell bit in the previous clause.
And then the code between 296 and 301 is identical to the Haskell bit except for the Javascript/Haskell atom, so it could be extracted as a generic function "sendToSimulation source contract" or something like that common to both Javascript and Haskell